### PR TITLE
Implement symbol wildcard

### PR DESCRIPTION
**Pull request by Werdor**
Feature : symbol wildcard (only for market = */BTC)

```
"crypto-currencies": {
        "Bitcoin": {
            "pairs": "*",
            "name": "BTC",
            "add": ["BTC/USDT"] // optional
        }
    },
```

https://github.com/Drakkar-Software/OctoBot/wiki/Configuration#wildcard

### CODE REVIEW

File: core/evaluator_factory.py [hunk 1]
```diff
@@ -50,46 +50,48 @@ def create(self):
...
 
     def create_dispatchers(self):
-        self.dispatchers_list = DispatcherCreator.create_dispatchers(self.octobot.get_config(), self.octobot.get_async_loop())
+        self.dispatchers_list = DispatcherCreator.create_dispatchers(self.octobot.get_config(),
+                                                                     self.octobot.get_async_loop())
 
     def evaluation_tasks_creation(self):
         self.logger.info("Evaluation threads creation...")
 
+        crypto_currency_list = self.octobot.get_config()[CONFIG_CRYPTO_CURRENCIES].items()
```
Comment (GuillaumeDSM) [inline core/evaluator_factory.py]: why ```.items()``` and not .keys() and then simply ```for crypto_currency in crypto_currency_list:``` ?

File: tests/unit_tests/trading_tests/test_exchange_manager.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        exchange_manager._set_config_traded_pairs()
+
+        assert "ICX/BTC" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "NEO/BTC" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "VEN/BTC" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "XLM/BTC" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "ONT/BTC" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "BTC/USDT" in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "ETH/USDT" not in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
+        assert "NEO/BNB" not in exchange_manager.cryptocurrencies_traded_pairs["Bitcoin"]
```
Comment (GuillaumeDSM) [inline tests/unit_tests/trading_tests/test_exchange_manager.py]: 👍

File: core/evaluator_factory.py (hunk 1 above)
Comment (Werdor) [reply core/evaluator_factory.py]: Exactly :+1:

**Comment by Werdor** [comment]
> > Feature : symbol wildcard (only for market = */BTC)
> > ```
> > "crypto-currencies": {
> >         "Bitcoin": {
> >             "pairs": "*",
> >             "name": "BTC",
> >             "add": ["BTC/USDT"] // optional
> >         }
> >     },
> > ```
> > https://github.com/Drakkar-Software/OctoBot/wiki/Configuration#wildcard
> 
> This means it will listen to all pairs having BTC as quote ?
> In this case I think the wildcard symbol should not be _*_ but something like _ALL_WITH_NAME_AS_MARKET_

You mean replacing `*` by a string like `ALL_WITH_NAME_AS_MARKET` ?

**Comment by Werdor** [comment]
> > > Feature : symbol wildcard (only for market = */BTC)
> > > ```
> > > "crypto-currencies": {
> > >         "Bitcoin": {
> > >             "pairs": "*",
> > >             "name": "BTC",
> > >             "add": ["BTC/USDT"] // optional
> > >         }
> > >     },
> > > ```
> > > https://github.com/Drakkar-Software/OctoBot/wiki/Configuration#wildcard
> > 
> > 
> > This means it will listen to all pairs having BTC as quote ?
> > In this case I think the wildcard symbol should not be _*_ but something like _ALL_WITH_NAME_AS_MARKET_
> 
> You mean replacing `*` by a string like `ALL_WITH_NAME_AS_MARKET` ?

Because I think `*` (wildcard) is more comprehensive than a constant name as string.

**Comment by GuillaumeDSM** [comment]
> > > > Feature : symbol wildcard (only for market = */BTC)
> > > > ```
> > > > "crypto-currencies": {
> > > >         "Bitcoin": {
> > > >             "pairs": "*",
> > > >             "name": "BTC",
> > > >             "add": ["BTC/USDT"] // optional
> > > >         }
> > > >     },
> > > > ```
> > > > https://github.com/Drakkar-Software/OctoBot/wiki/Configuration#wildcard
> > > 
> > > 
> > > This means it will listen to all pairs having BTC as quote ?
> > > In this case I think the wildcard symbol should not be _*_ but something like _ALL_WITH_NAME_AS_MARKET_
> > 
> > 
> > You mean replacing `*` by a string like `ALL_WITH_NAME_AS_MARKET` ?
> 
> Because I think `*` (wildcard) is more comprehensive than a constant name as string.
> 
> >  GitHub**[Drakkar-Software/OctoBot](https://github.com/Drakkar-Software/OctoBot)**Cryptocurrency trading bot: high frequency, daily trading, social trading, ... - Drakkar-Software/OctoBot

If we leave *, it will tell users that it will look for all pairs containing "name" either in quote or base which is not the case.
To explain it will only look for pairs with "name" as base and not all pair including "name" as quote yes I think something like *ALL_WITH_NAME_AS_MARKET* can be clearer, what do you think ?

**Comment by Werdor** [comment]
> > > > > Feature : symbol wildcard (only for market = */BTC)
> > > > > ```
> > > > > "crypto-currencies": {
> > > > >         "Bitcoin": {
> > > > >             "pairs": "*",
> > > > >             "name": "BTC",
> > > > >             "add": ["BTC/USDT"] // optional
> > > > >         }
> > > > >     },
> > > > > ```
> > > > > https://github.com/Drakkar-Software/OctoBot/wiki/Configuration#wildcard
> > > > 
> > > > 
> > > > This means it will listen to all pairs having BTC as quote ?
> > > > In this case I think the wildcard symbol should not be _*_ but something like _ALL_WITH_NAME_AS_MARKET_
> > > 
> > > 
> > > You mean replacing `*` by a string like `ALL_WITH_NAME_AS_MARKET` ?
> > 
> > 
> > Because I think `*` (wildcard) is more comprehensive than a constant name as string.
> > >  GitHub**[Drakkar-Software/OctoBot](https://github.com/Drakkar-Software/OctoBot)**Cryptocurrency trading bot: high frequency, daily trading, social trading, ... - Drakkar-Software/OctoBot
> 
> If we leave *, it will tell users that it will look for all pairs containing "name" either in quote or base which is not the case.
> To explain it will only look for pairs with "name" as base and not all pair including "name" as quote yes I think something like _ALL_WITH_NAME_AS_MARKET_ can be clearer, what do you think ?
> 
> >  GitHub**[Drakkar-Software/OctoBot](https://github.com/Drakkar-Software/OctoBot)**Cryptocurrency trading bot: high frequency, daily trading, social trading, ... - Drakkar-Software/OctoBot

I don't think a constant name as a value in config is a good idea. To solve your issue I think we can replace the key `name` with `quote`, what do you think ?

**Comment by GuillaumeDSM** [comment]
> > If we leave *, it will tell users that it will look for all pairs containing "name" either in quote or base which is not the case.
> > To explain it will only look for pairs with "name" as base and not all pair including "name" as quote yes I think something like _ALL_WITH_NAME_AS_MARKET_ can be clearer, what do you think ?
> 
> I don't think a constant name as a value in config is a good idea. To solve your issue I think we can replace the key `name` with `quote`, what do you think ?

Excellent idea, much better yes !

**Pull request merged**